### PR TITLE
10.0 checkout compatibility oca suggest create account

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: website_sale_checkout_store/controllers/main.py [hunk 1]
```diff
@@ -9,20 +9,22 @@ class WebsiteSaleExtended(WebsiteSale):
     @http.route(['/shop/checkout'], type='http', auth="public", website=True)
     def checkout(self, **post):
         order = request.website.sale_get_order()
+        # order.checkout = False
```
Comment (yelizariev) [inline website_sale_checkout_store/controllers/main.py]: remove commented code

File: website_sale_checkout_store/controllers/main.py [hunk 2]
```diff
@@ -50,13 +53,17 @@ def payment(self, **post):
     @http.route('/shop/payment/get_status/<int:sale_order_id>', type='json', auth="public", website=True)
     def payment_get_status(self, sale_order_id, **post):
         order = request.env['sale.order'].sudo().browse(sale_order_id)
+        if type(order.buy_way) is bool:
```
Comment (yelizariev) [inline website_sale_checkout_store/controllers/main.py]: Can you use following instead?

     if not order.buy_way

File: website_sale_checkout_store/controllers/main.py [hunk 3]
```diff
@@ -50,13 +53,17 @@ def payment(self, **post):
...
+            return super(WebsiteSaleExtended, self).payment_get_status(sale_order_id, **post)
         if 'nobill' in order.buy_way:
             return {'recall': False, 'message': ''}
         else:
             return super(WebsiteSaleExtended, self).payment_get_status(sale_order_id, **post)
 
     def _get_mandatory_fields(self):
         order = request.website.sale_get_order()
+        if type(order.buy_way) is bool:
+            return ["name", "phone", "email", "street", "city", "country_id"]
```
Comment (yelizariev) [inline website_sale_checkout_store/controllers/main.py]: This case is already covered in latest case below

File: website_sale_checkout_store/templates.xml [hunk 4]
```diff
@@ -14,69 +14,78 @@
...
-                        <a t-if="nobill_ship and not optional_products and website_sale_order and website_sale_order.website_order_line"
-                           id="nobill_ship" class="btn btn-primary pull-right"
-                           href="/shop/checkout?buyMethod=nobill_ship" style="width: 100%; margin-bottom: 5px">
-                            <span class="hidden-xs">Pay on delivery</span>
-                            <span class="visible-xs-inline">Pay on delivery</span>
-                            <span class="fa fa-long-arrow-right"/>
-                        </a>
-                    </div>
-                </div>
+                </t>
```
Comment (yelizariev) [inline website_sale_checkout_store/templates.xml]: Don't change indent please

File: website_sale_checkout_store/controllers/main.py (hunk 2 above)
Comment (KolushovAlexandr) [reply website_sale_checkout_store/controllers/main.py]: it should be executed in both options: true and false

File: website_sale_checkout_store/controllers/main.py (hunk 1 above)
Comment (KolushovAlexandr) [reply website_sale_checkout_store/controllers/main.py]: i've got that commit in pr 246, from which i made the branch, but idk why here this commit is absent

File: website_sale_checkout_store/controllers/main.py (hunk 2 above)
Comment (yelizariev) [reply website_sale_checkout_store/controllers/main.py]: ok

Comment (yelizariev) [reply website_sale_checkout_store/controllers/main.py]: How it can be True?

File: website_sale_checkout_store/controllers/main.py [hunk 5]
```diff
@@ -9,37 +9,32 @@ class WebsiteSaleExtended(WebsiteSale):
...
-            if not order.partner_id[f]:
-                return request.redirect('/shop/address?partner_id=%d' % order.partner_id.id)
+        if order.buy_way:
+            if order.partner_id.id == request.website.user_id.sudo().partner_id.id:
+                return request.redirect('/shop/address')
+            sale_order_id = request.session.get('sale_order_id')
+            if 'noship' in order.buy_way and 'nobill' in order.buy_way:
+                request.session['sale_last_order_id'] = order.id
+                request.website.sale_reset()
+                return request.redirect('/shop/confirmation')
```
Comment (yelizariev) [inline website_sale_checkout_store/controllers/main.py]: Can we make those redirection via extending ``checkout_redirection`` and just call super?

Comment (yelizariev) [reply website_sale_checkout_store/controllers/main.py]: Canceled

File: website_sale_checkout_store/templates.xml [hunk 6]
```diff
@@ -14,6 +14,8 @@
         </template>
         <template id="website_sale_checkout_store_cart" name="my_cart" inherit_id="website_sale.cart" priority="50">
             <xpath expr="//div[@class='row']//a[@class='btn btn-primary pull-right mb32'][@href='/shop/checkout']" position="replace">
+              <t t-if="hide_checkout_button is None or hide_checkout_button">
+                <t t-set="button_class" t-value="(suggest_create_account is None or suggest_create_account) and 'btn btn-primary pull-right' or 'btn btn-default pull-right'"/>
```
Comment (yelizariev) [inline website_sale_checkout_store/templates.xml]: Suspicious computation. Please check

    (suggest_create_account is None or suggest_create_account) and 'btn btn-primary pull-right' or 'btn btn-default pull-right'

As I understand, It should be ``suggest_create_account and  'btn btn-default pull-right' or 'btn btn-primary pull-right'``

Comment (KolushovAlexandr) [reply website_sale_checkout_store/templates.xml]: variables names confuse a bit. 
suggest_create_account is None -> module isnt installed
suggest_create_account is True -> doesnt allow uninvited to signup
suggest_create_account is False ->  allow uninvited to signup
If it is impossible to sign up -> primary (blue) class
If it is possible to sign up -> default (white) class to suggest customer to sign up

Comment (yelizariev) [reply website_sale_checkout_store/templates.xml]: This statement is wrong

     suggest_create_account is True -> doesnt allow uninvited to signup

Comment (yelizariev) [reply website_sale_checkout_store/templates.xml]: If the statement is correct, then it has to be renamed or fixed in OCA modules

**Comment by yelizariev** [comment]
default

**Pull request merged**